interfaces/builtin/home.go

```diff
@@ -107,16 +110,25 @@ func (iface *homeInterface) BeforePreparePlug(plug *snap.PlugInfo) error {
 	return nil
 }
 
+func evalPrompting(s string) string {
+	repl := ""
+	if features.AppArmorPrompting.IsEnabled() && apparmor_prompting.SupportAvailable() {
```

Comment:
if we make the feature unexported we need to think how to have the information reach here

Reply:
The parser also needs to be probed to ensure it supports the prompt syntax - currently `SupportAvailable()` only checks the kernel via the presence of a specific file - ideally instead of that we would use `KernelFeatures()` to indicate that the kernel supports prompting and so `SupportAvailable()` could be removed entirely.

Reply:
I'll look into it, thanks! Is there any reason not to have `SupportAvailable()` checked as well? The prompting listener only runs if `SupportAvailable()`, though it could be adjusted to check `KernelFeatures` as well. I think I want to keep using at least `SupportAvailable()` in the listener, since it tries to open the socket it's checking for.

Reply:
Yes I think this makes sense - keep `SupportAvailable()` but have it check both `KernelFeatures` and `ParserFeatures`

Reply:
Sounds good, thanks!

Digging around a bit in the kernel features, I don't see anything related to prompting in `/sys/kernel/security/apparmor/features`:
```
$ tree /sys/kernel/security/apparmor/features 
/sys/kernel/security/apparmor/features
├── capability
├── caps
│   └── mask
├── dbus
│   └── mask
├── domain
│   ├── attach_conditions
│   │   └── xattr
│   ├── change_hat
│   ├── change_hatv
│   ├── change_onexec
│   ├── change_profile
│   ├── computed_longest_left
│   ├── fix_binfmt_elf_mmap
│   ├── post_nnp_subset
│   ├── stack
│   └── version
├── file
│   └── mask
├── io_uring
│   └── mask
├── ipc
│   └── posix_mqueue
├── mount
│   └── mask
├── namespaces
│   ├── mask
│   ├── pivot_root
│   └── profile
├── network
│   ├── af_mask
│   └── af_unix
├── network_v8
│   └── af_mask
├── policy
│   ├── outofband
│   ├── permstable32
│   ├── permstable32_version
│   ├── set_load
│   ├── unconfined_restrictions
│   │   ├── change_profile
│   │   ├── io_uring
│   │   └── userns
│   └── versions
│       ├── v5
│       ├── v6
│       ├── v7
│       ├── v8
│       └── v9
├── ptrace
│   └── mask
├── query
│   └── label
│       ├── data
│       ├── multi_transaction
│       └── perms
├── rlimit
│   └── mask
└── signal
    └── mask
```
This is on kernel `6.5.0-21-generic #21-Ubuntu`, which should support prompting. From what I can see, kernel features are computed based on directories here. https://github.com/snapcore/snapd/blob/master/sandbox/apparmor/apparmor.go#L338

Is prompting covered by one of these features, or is it not yet listed in the supported kernel features?

Reply:
Apologies, I had assumed there was a particular feature file - instead we would need to parse `/sys/kernel/security/apparmor/features/policy/permstable32` and look for the string `prompt` within it - but perhaps it will be enough to just add support for probing this via `ParserFeatures()` - I am hoping that in the case that the kernel doesn't support the feature that the parser will raise an error.

Reply:
No worries, thanks for clarifying. I see `prompt` in `/v2/kernel/security/apparmor/features/policy/permstable32`, so that should be great, thank you. I'll look around if there's an existing helper function to parse and check this file.

I see that `ParserFeatures()` iterates through a list of features and checks each one with a probe string and optional flags. Do you know what these names/flags should be for prompting? I assume `prompt`, but not sure about any flags which might be required.

Reply:
So since `permstable32` is included in `KernelFeatures()` we may want to consider reading that file whilst parsing the existing features and exposing the values within `KernelFeatures()` somehow - then you could just check if a feature like `policy:perstable32:prompt` existing in `KernelFeatures()`.

For `ParserFeatures()`, I wouldn't worry about flags and just probe with a string like `prompt /foo r,` or somesuch - that should be sufficient I think.

Reply:
Not strictly necessary for this PR, but what about `change_profile`, `io_uring`, and `userns` under `unconfined_restrictions`? Are those kernel features as well, and should they be included in the list? It seems like some features are not directory names -- sometimes file names (as those listed above), sometimes features listed in a file (e.g. permstable32`).

Reply:
Thanks for the suggestions! The latest commit has proper parser and kernel checking, and the feature is no longer exported.